client/blocks/edit-gravatar/test/index.jsx

```diff
@@ -37,172 +78,147 @@ describe( 'EditGravatar', () => {
 
-			expect( wrapper.update().find( ImageEditor ).length ).toEqual( 0 );
+			expect( screen.queryByTestId( 'image-editor' ) ).not.toBeInTheDocument();
 			expect( uploadGravatarSpy ).toHaveBeenCalledTimes( 1 );
-			expect( receiveGravatarImageFailedSpy.mock.calls ).toHaveLength( 0 );
+			expect( receiveGravatarImageFailedSpy ).not.toHaveBeenCalled();
 		} );
 
-		test( 'given an error, hides image editor and calls error notice action creator', () => {
-			const error = new Error();
+		test( 'given an error, hides image editor and calls error notice action creator', async () => {
+			const user = userEvent.setup();
 			const receiveGravatarImageFailedSpy = jest.fn();
 			const uploadGravatarSpy = jest.fn();
-			const wrapper = shallow(
+			render(
 				<EditGravatar
+					{ ...props }
 					receiveGravatarImageFailed={ receiveGravatarImageFailedSpy }
-					resetAllImageEditorState={ noop }
-					translate={ noop }
 					uploadGravatar={ uploadGravatarSpy }
-					user={ user }
-					recordReceiveImageEvent={ noop }
 				/>
 			);
 
-			wrapper.instance().onReceiveFile( files );
-			wrapper.instance().onImageEditorDone( error, 'image', imageEditorProps );
+			ImageEditor.mockImplementationOnce( ( { onDone } ) => (
+				<div data-testid="image-editor">
+					<button
+						data-testid="image-editor-done"
+						onClick={ () => onDone( 'some arbitrary error occured' ) }
+					>
+						Done
+					</button>
+				</div>
+			) );
 
-			expect( wrapper.update().find( ImageEditor ).length ).toEqual( 0 );
+			const fileInput = screen.queryByTestId( 'file-picker-input' );
+			await user.upload( fileInput, files );
+
+			expect( screen.queryByTestId( 'image-editor' ) ).toBeVisible();
+			await user.click( screen.getByTestId( 'image-editor-done' ) );
+
+			expect( screen.queryByTestId( 'image-editor' ) ).not.toBeInTheDocument();
 			expect( receiveGravatarImageFailedSpy ).toHaveBeenCalledTimes( 1 );
-			expect( uploadGravatarSpy.mock.calls ).toHaveLength( 0 );
+			expect( uploadGravatarSpy ).not.toHaveBeenCalled();
 		} );
 	} );
 
 	describe( 'unverified user', () => {
-		test( 'shows email verification dialog when clicked', () => {
-			const wrapper = shallow(
-				<EditGravatar translate={ noop } user={ user } recordClickButtonEvent={ noop } />
-			);
-			// Enzyme requires simulate() to be called directly on the element with the click handler
-			const clickableWrapper = wrapper.find( '.edit-gravatar > div' ).first();
+		test( 'shows email verification dialog when clicked', async () => {
+			const user = userEvent.setup();
+			const { container } = render( <EditGravatar { ...props } /> );
+
+			expect( screen.queryByTestId( 'verify-email-dialog' ) ).not.toBeInTheDocument();
 
-			clickableWrapper.simulate( 'click' );
-			wrapper.update(); // make sure the state has been updated
-			expect( wrapper.find( VerifyEmailDialog ) ).toHaveLength( 1 );
+			await user.click( container.firstChild.firstChild );
```

Comment:
It's a wrapping `<div />` element which has an added `onClick` handler. I don't see an accessible way to query it because the element isn't accessible (preexisting). Would need to adjust the component and move that listener to a more accessible place. Would you mind me following up on that one?

Reply:
Sounds good to me 👍 Let's follow up in another PR.